oss-internship-2020/libarchive/examples/sapi_minitar_main.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file contains the main function from the original minitar example:
+// https://github.com/libarchive/libarchive/blob/master/examples/minitar/minitar.c
+// Most of the logic is the same, it was only simplified a bit since this is
+// only used for the command line tool.
+// No sandboxing takes place in this function.
+
+#include <iostream>
+
+#include "sapi_minitar.h"
+
+static void usage(void);
+
+int main(int argc, const char** argv) {
+  google::InitGoogleLogging(argv[0]);
+  const char* filename = NULL;
+  int compress, flags, mode, opt;
+
+  (void)argc;
```

Comment:
If `argc` really is unused, rename it to `unused_argc` to silence linter warnings or remove the name from the parameter list.